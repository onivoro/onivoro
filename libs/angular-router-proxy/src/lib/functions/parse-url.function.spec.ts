import { parseUrl } from './parse-url.function';

describe('parseUrl', () => {
    it.each([
        'items/',
        'items',
        'items/:id',
        '/',
        '',
    ])('parses url into segments and param map given url %j', (url: string) => {
        expect(parseUrl(url)).toMatchSnapshot();
    });
});