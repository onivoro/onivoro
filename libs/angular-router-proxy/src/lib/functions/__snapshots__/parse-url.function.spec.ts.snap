// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseUrl parses url into segments and param map given url "" 1`] = `
Object {
  "params": Object {},
  "segments": Array [
    "",
  ],
}
`;

exports[`parseUrl parses url into segments and param map given url "/" 1`] = `
Object {
  "params": Object {},
  "segments": Array [
    "",
  ],
}
`;

exports[`parseUrl parses url into segments and param map given url "items" 1`] = `
Object {
  "params": Object {},
  "segments": Array [
    "items",
  ],
}
`;

exports[`parseUrl parses url into segments and param map given url "items/" 1`] = `
Object {
  "params": Object {},
  "segments": Array [
    "items",
  ],
}
`;

exports[`parseUrl parses url into segments and param map given url "items/:id" 1`] = `
Object {
  "params": Object {
    ":id": "",
  },
  "segments": Array [
    "items",
    ":id",
  ],
}
`;
